src/sentry/integrations/source_code_management/commit_context.py

```diff
@@ -94,16 +100,28 @@ def get_blame_for_files(
 
         files: list of FileBlameInfo objects
         """
-        try:
-            client = self.get_client()
-        except Identity.DoesNotExist:
-            return []
-        try:
-            response = client.get_blame_for_files(files, extra)
-        except IdentityNotValid:
-            return []
-
-        return response
+        with CommitContextIntegrationInteractionEvent(
+            interaction_type=SCMIntegrationInteractionType.GET_BLAME_FOR_FILES,
+            provider_key=self.integration_name,
+        ).capture() as lifecycle:
+            try:
+                client = self.get_client()
+            except Identity.DoesNotExist as e:
+                lifecycle.record_failure(e)
+                sentry_sdk.capture_exception(e)
+                return []
+            try:
+                response = client.get_blame_for_files(files, extra)
+            except IdentityNotValid as e:
+                lifecycle.record_failure(e)
```

Comment:
should we ask Issues team why these specific errors are caught? i'm not sure if this is a failure on our side or the user's side. it seems like this happens when refreshing the identity fails

Reply:
let me do that. i think we can record_failure to see how often this happens then adjust?